src/metadataGeneration/parameterGenerator.ts

```diff
@@ -110,27 +110,35 @@ export class ParameterGenerator {
     const parameterName = (parameter.name as ts.Identifier).text;
     const type = this.getValidatedType(parameter, false);
 
+    const commonProperties = {
+      default: getInitializerValue(parameter.initializer, type),
+      description: this.getParameterDescription(parameter),
+      in: 'query' as 'query',
```

Comment:
What does this do?

Reply:
This because `in` property is typed as `'query' | 'header' | 'path' | 'formData' | 'body' | 'body-prop' | 'request'`. With just `in: 'query'` for the definition, it is typed as `string` by default, so there is an incompatible type error when returning the parameter.
An alternative would be to use `as Tsoa.Parameter` for the return.